src/reactions/ReactionListTabView.js

```diff
@@ -0,0 +1,49 @@
+/* @flow strict-local */
+import React from 'react';
+import { Dimensions, View } from 'react-native';
+import { TabNavigator, TabBarTop } from 'react-navigation';
+
+import type { AggregatedReaction } from '../types';
+import ReactionUserList from './ReactionUserList';
+import tabsOptions from '../styles/tabs';
+import { RawLabel } from '../common';
+import EmojiRow from '../emoji/EmojiRow';
+import styles from '../styles';
+
+export default (
+  aggregatedReactions: $ReadOnlyArray<AggregatedReaction>,
+  initialRouteName: string,
+) => {
+  const reactionsTabs = aggregatedReactions.reduce((tabs, aggregatedReaction) => {
+    const { width } = Dimensions.get('window');
+    const emojiSize: number = Math.min((width / aggregatedReactions.length) * 0.32, 24);
+    const countSize: number = Math.min((width / aggregatedReactions.length) * 0.4, 16);
```

Comment:
This looks like it will make things giant when there's just one or two reactions. Will it, or is there something I'm missing?

Reply:
There was indeed something I was missing -- I was reading `Math.min` the wrong way around. :slightly_smiling_face: That's taken care of, then.

On the other hand we should also have the thing that I *thought* these were doing -- we should have minimum values so that when there are a *lot* of reactions, they don't become too tiny to read. Better to show an incomplete list that can be read, than a complete list where nothing can be read.

Stepping back a level (and as we discussed by voice today): instead of doing this computation based on `Dimensions.get('window')`, it'd be much cleaner and more robust to do it with flexbox style props. I know CSS flexbox has ample features to do what we want here, and hopefully RN's flexbox does too. As is, this version makes an assumption that the component has the whole width of the window, and that makes it more fragile than it would be with flexbox layout.

Reply:
For now, let's drop this commit. 

Actually we don't even want `flex` here. React navigation has its own measurement. It make sure that tab bar occupy whole space available to it. And tabs content is in centre. 

Add minWidth, maxWidth doesn't work as expected. 

Anyway it is rare case when messages have so many reactions. Let's work on this on another PR.